pkg/api/validation/validation.go

```diff
@@ -522,6 +527,22 @@ func validateGlusterfs(glusterfs *api.GlusterfsVolumeSource) errs.ValidationErro
 	return allErrs
 }
 
+func validateFlocker(flocker *api.FlockerVolumeSource) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+	if flocker.DatasetName == "" {
+		allErrs = append(allErrs, errs.NewFieldRequired("datasetName"))
+	}
+	if strings.Contains(flocker.DatasetName, "/") {
+		allErrs = append(allErrs, errs.NewFieldInvalid("datasetName", flocker.DatasetName, "must not contain '/'"))
+	}
+	if flocker.Size != "" {
+		if _, err := units.FromHumanSize(flocker.Size); err != nil {
```

Comment:
Oh cool, I am going to take a look to that one. Sorry, I didn't know it exists and I checked the Godeps to see what you had there.